packages/block-library/src/button/save.js

```diff
@@ -45,17 +45,19 @@ export default function save( { attributes, className } ) {
 	} );
 
 	return (
-		<div { ...useBlockProps.save( { className: wrapperClasses } ) }>
-			<RichText.Content
-				tagName="a"
-				className={ buttonClasses }
-				href={ url }
-				title={ title }
-				style={ buttonStyle }
-				value={ text }
-				target={ linkTarget }
-				rel={ rel }
-			/>
-		</div>
+		text && (
```

Comment:
Nitpick: you could have sed and early return here. Not that it changes anything, just wanted to share this code style trick in case you find it nicer.

Reply:
Thank you!
I considered that when I saw the previous PR, but when I tried it, I got a lot of warnings/errors regarding unused variables (from everything included in the return that was skipped).

Is there a way to work around that, for future knowledge at least? I think this is pretty readable, but an early return is probably slightly better.

Reply:
The answer is to just move the early return above the unused variables. I went ahead and made a PR for this: #29781.

Reply:
That PR is good. Though I would like to add just a small thing here. Early returns in components can break the "react hooks rules", so if we use this technique, we should make sure there's no React hook being called after the early return.

Reply:
@youknowriad Good point. That kind of thing is caught by the React `rules-of-hooks` ESLint rules, isn't it? Or are we not using that?

Reply:
you're probably right :)

Reply:
@youknowriad that rule doesn't apply to custom hooks like `useBlockProps`?

Just want to know for future reference.

Reply:
@Mamaduka `useBlockProps.save()` is not a hook. In fact, React hooks aren't even usable in a block's save implementation. It's actually just an alias that points to `getBlockProps()` from https://github.com/WordPress/gutenberg/blob/trunk/packages/blocks/src/api/serializer.js. Yeah, it's kinda weird.

Reply:
Thanks, @ZebulanStanphill for clarifying this.